File: PHP/Medium/Sorting/1305__All_Elements_in_Two_Binary_Search_Trees.php
<?php

class TreeNode
{
    public function __construct(public $val = 0, public $left = null, public $right = null)
    {
    }
}

class Solution
{
    function inorderTraversal($root, &$result)
    {
        if ($root === null) {
            return;
        }
        $this->inorderTraversal($root->left, $result);
        $result[] = $root->val;
        $this->inorderTraversal($root->right, $result);
    }

    function mergeSortedLists($list1, $list2)
    {
        $merged = [];
        $i = 0;
        $j = 0;
        while ($i < count($list1) && $j < count($list2)) {
            if ($list1[$i] < $list2[$j]) {
                $merged[] = $list1[$i++];
            } else {
                $merged[] = $list2[$j++];
            }
        }
        while ($i < count($list1)) {
            $merged[] = $list1[$i++];
        }
        while ($j < count($list2)) {
            $merged[] = $list2[$j++];
        }
        return $merged;
    }

    function getAllElements($root1, $root2)
    {
        $list1 = [];
        $list2 = [];
        $this->inorderTraversal($root1, $list1);
        $this->inorderTraversal($root2, $list2);

        return $this->mergeSortedLists($list1, $list2);
    }
}

File: Python/Medium/Sorting/1305__All_Elements_in_Two_Binary_Search_Trees.py
from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def inorder_traversal(self, root: Optional[TreeNode], result: List[int]) -> None:
        if root:
            self.inorder_traversal(root.left, result)
            result.append(root.val)
            self.inorder_traversal(root.right, result)

    def getAllElements(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> List[int]:
        list1, list2 = [], []
        self.inorder_traversal(root1, list1)
        self.inorder_traversal(root2, list2)

        # Merge two sorted lists
        merged = []
        i, j = 0, 0
        while i < len(list1) and j < len(list2):
            if list1[i] < list2[j]:
                merged.append(list1[i])
                i += 1
            else:
                merged.append(list2[j])
                j += 1

        while i < len(list1):
            merged.append(list1[i])
            i += 1
        while j < len(list2):
            merged.append(list2[j])
            j += 1

        return merged
